### PR TITLE
Remove os checks from initialize

### PR DESCRIPTION
**Pull request by miah**
Removes `skip_resource` and `raise .. if InSpec.os` stuff from initialize as this is covered by platform support.

Signed-off-by: Miah Johnson <miah@chia-pet.org>

Fixes https://github.com/chef/inspec/issues/2714

### CODE REVIEW

File: lib/resources/aide_conf.rb [hunk 1]
```diff
@@ -27,7 +27,6 @@ class AideConf < Inspec.resource(1)
     include CommentParser
 
     def initialize(aide_conf_path = nil)
-      return skip_resource 'The `aide_conf` resource is not supported on your OS.' unless inspec.os.linux?
```
Comment (jquick) [inline lib/resources/aide_conf.rb:30]: We should make the supports linux here to match.

File: lib/resources/etc_fstab.rb [hunk 2]
```diff
@@ -28,7 +28,6 @@ class EtcFstab < Inspec.resource(1)
     include CommentParser
 
     def initialize(fstab_path = nil)
-      return skip_resource 'The `etc_fstab` resource is not supported on your OS.' unless inspec.os.linux?
```
Comment (jquick) [inline lib/resources/etc_fstab.rb:31]: Same as above.

File: lib/resources/etc_hosts_allow_deny.rb [hunk 3]
```diff
@@ -20,7 +20,6 @@ class EtcHostsAllow < Inspec.resource(1)
     include CommentParser
 
     def initialize(hosts_allow_path = nil)
-      return skip_resource 'The `etc_hosts_allow` resource is not supported on your OS.' unless inspec.os.linux?
```
Comment (jquick) [inline lib/resources/etc_hosts_allow_deny.rb:23]: Same as above.

File: lib/resources/postgres_hba_conf.rb [hunk 4]
```diff
@@ -18,7 +18,6 @@ class PostgresHbaConf < Inspec.resource(1)
 
     # @todo add checks to ensure that we have data in our file
     def initialize(hba_conf_path = nil)
-      return skip_resource 'The `postgres_hba_conf` resource is not supported on your OS.' unless inspec.os.linux?
```
Comment (jquick) [inline lib/resources/postgres_hba_conf.rb:21]: Same as above.

File: lib/resources/postgres_ident_conf.rb [hunk 5]
```diff
@@ -17,12 +17,10 @@ class PostgresIdentConf < Inspec.resource(1)
     attr_reader :params, :conf_file
 
     def initialize(ident_conf_path = nil)
-      return skip_resource 'The `postgres_ident_conf` resource is not supported on your OS.' unless inspec.os.linux?
```
Comment (jquick) [inline lib/resources/postgres_ident_conf.rb:20]: Same as above.

**Review by jquick — DISMISSED** [review]
Thanks @miah ! Looking good. Just a few notes.

File: lib/resources/aide_conf.rb (hunk 1 above)
Comment (miah) [reply lib/resources/aide_conf.rb:30]: The reason I left these as UNIX is they are fairly generic and should work across all UNIX systems.

**Comment by miah** [comment]
@jquick The reason I left these as UNIX is they are fairly generic and should work across all UNIX systems.

**Review by jquick — APPROVED** [review]
Sounds good! Thanks @miah .

**Review by jerryaldrichiii — APPROVED** [review]
Looks good to me, thanks @miah!

**Pull request merged**